internal/provider/data_source_gitlab_project_cluster_agent_test.go

```diff
@@ -0,0 +1,60 @@
+package provider
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestAccDataSourceGitlabProjectClusterAgent_basic(t *testing.T) {
+	testAccCheck(t)
+
+	testProject := testAccCreateProject(t)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: providerFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: fmt.Sprintf(`
+					resource "gitlab_project_cluster_agent" "this" {
+						project           = "%d"
+						name              = "agent-1"
+					}
+
+					data "gitlab_project_cluster_agent" "this" {
+						project           = gitlab_project_cluster_agent.this.project
+						agent_id          = gitlab_project_cluster_agent.this.agent_id
+					}
+					`, testProject.ID,
+				),
+				Check: resource.ComposeTestCheckFunc(
+					testAccDataSourceGitlabProjectClusterAgent("gitlab_project_cluster_agent.this", "data.gitlab_project_cluster_agent.this"),
+				),
```

Comment:
@timofurrer  - I'm a little confused here, the `ComposeTestCheckFunc` is meant to combine multiple Test Functions into one, but there is only one test function here. Is there a reason to use the `ComposeTestCheckFunc` here?

Side note, I think I wrote a helper for this in a previous PR that we should probably deprecate/remove. Whoops!

Reply:
Copy & Paste mistake :D 

changed it anyways here.